src/meds_torch/schemas/generate_analysis_schema.py

```diff
@@ -0,0 +1,124 @@
+# src/meds_torch/schemas/generate_analysis_schema.py
+
+from collections import OrderedDict
+
+import polars as pl
+import pyarrow as pa
+
+from meds_torch.models import ACTUAL_FUTURE, GENERATE_PREFIX, INPUT_DATA
+
+# Define struct fields in a fixed order using OrderedDict
+TRAJECTORY_FIELDS = OrderedDict(
+    [
+        ("code", pa.list_(pa.int64())),
+        ("mask", pa.list_(pa.bool_())),
+        ("numeric_value", pa.list_(pa.float64())),
+        ("numeric_value_mask", pa.list_(pa.bool_())),
+        ("time_delta_days", pa.list_(pa.float64())),
+    ]
+)
+
+# Create trajectory_data_type with fixed field order
+trajectory_data_type = pa.struct([(name, dtype) for name, dtype in TRAJECTORY_FIELDS.items()])
+
+
+def generation_analysis_schema(num_samples=0, do_include_actual=True):
+    """
+    Generate the schema for trajectory analysis data with consistent field ordering.
+
+    Args:
+        num_samples (int): Number of generated samples
+        do_include_actual (bool): Whether to include actual future data
+
+    Returns:
+        pyarrow.Schema: The schema for trajectory analysis
+    """
+    # Use OrderedDict to maintain consistent field order
+    schema_fields = OrderedDict(
+        [
+            ("subject_id", pa.int64()),
+            ("prediction_time", pa.timestamp("us")),
+            (INPUT_DATA, trajectory_data_type),
+        ]
+    )
+
+    # Add generated sample fields in order
+    for i in range(num_samples):
+        schema_fields[f"{GENERATE_PREFIX}{i}"] = trajectory_data_type
```

Comment:
This doesn't seem like what you want... why not use a longitudinal encoding instead of a wide encoding?

Reply:
Ahhh I see, you're saying we could convert to meds-like data here instead?

Reply:
The pros of storing the raw tensors is that it may help with debugging zero-shot classification tasks that users develop on the raw tensorized data, but there is likely a better workflow for that.